### PR TITLE
to yarn we go

### LINKED ISSUE #6820: T - move to Yarn (developer experience)

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #6820

~Have to figure out where else things need to change. Most likely in the dev documentation.~
There was an old piece in the dev documentation about how to edit js dependencies and it was plainly wrong anyway.

### CODE REVIEW

**Comment by peterbe** [comment]
There are bunch of packages that can be upgraded but that can wait. The newrelic one had to be upgraded now because of Node engine version stuff.

File: package.json [hunk 1]
```diff
@@ -16,24 +16,24 @@
...
-        "webpack:web": "npm run webpack -- --env.presets web --env.presets svg",
-        "webpack:bcd": "npm run webpack -- --env.presets bcd --env.presets svg",
-        "webpack:node": "npm run webpack -- --env.presets node --env.presets svg",
-        "webpack:production:web": "npm run webpack:web -- --env.mode production",
-        "webpack:production:bcd": "npm run webpack:bcd -- --env.mode production",
-        "webpack:production:node": "npm run webpack:node -- --env.mode production",
-        "webpack:development:web": "npm run webpack:web -- --env.mode development --env.presets hardsource",
-        "webpack:development:bcd": "npm run webpack:bcd -- --env.mode development --env.presets hardsource",
-        "webpack:development:node": "npm run webpack:node -- --env.mode development --env.presets hardsource",
+        "debug:development:web": "yarn run debug -- --env.presets web --env.presets svg --env.mode development",
```
Comment (mindy) [inline package.json:19]: Looks like we can remove the extra ` -- ` in each of the scripts: 

```
kuma@032b1e308502:/app$ yarn run webpack:production:web -- --env.presets analyze
yarn run v1.22.4
warning From Yarn 1.0 onwards, scripts don't require "--" for options to be forwarded. In a future version, any explicit "--" will be forwarded as-is to the scripts.
```

File: docs/development.rst [hunk 2]
```diff
@@ -474,14 +474,30 @@ this approach the next time they're upgraded.
 
 Front-end toolchain dependencies
 --------------------------------
-The Front-end toolchain dependencies are managed by npm_, but not checked in to
-the repository. Follow these steps to add or upgrade a dependency:
+The Front-end toolchain dependencies are managed by yarn_, but not checked in to
```
Comment (mindy) [inline docs/development.rst:477]: Thanks for updating these instructions 👍

File: package.json (hunk 1 above)
Comment (mindy) [inline package.json:19]:
```suggestion
        "debug:development:web": "yarn run debug --env.presets web --env.presets svg --env.mode development",
```

File: package.json [hunk 3]
```diff
@@ -16,24 +16,24 @@
...
+        "webpack:web": "yarn run webpack -- --env.presets web --env.presets svg",
+        "webpack:bcd": "yarn run webpack -- --env.presets bcd --env.presets svg",
+        "webpack:node": "yarn run webpack -- --env.presets node --env.presets svg",
+        "webpack:production:web": "yarn run webpack:web -- --env.mode production",
+        "webpack:production:bcd": "yarn run webpack:bcd -- --env.mode production",
+        "webpack:production:node": "yarn run webpack:node -- --env.mode production",
+        "webpack:development:web": "yarn run webpack:web -- --env.mode development --env.presets hardsource",
+        "webpack:development:bcd": "yarn run webpack:bcd -- --env.mode development --env.presets hardsource",
+        "webpack:development:node": "yarn run webpack:node -- --env.mode development --env.presets hardsource",
```
Comment (mindy) [inline package.json:28]:
```suggestion
        "webpack:web": "yarn run webpack --env.presets web --env.presets svg",
        "webpack:bcd": "yarn run webpack --env.presets bcd --env.presets svg",
        "webpack:node": "yarn run webpack --env.presets node --env.presets svg",
        "webpack:production:web": "yarn run webpack:web --env.mode production",
        "webpack:production:bcd": "yarn run webpack:bcd --env.mode production",
        "webpack:production:node": "yarn run webpack:node --env.mode production",
        "webpack:development:web": "yarn run webpack:web --env.mode development --env.presets hardsource",
        "webpack:development:bcd": "yarn run webpack:bcd --env.mode development --env.presets hardsource",
        "webpack:development:node": "yarn run webpack:node --env.mode development --env.presets hardsource",
```

File: package.json [hunk 4]
```diff
@@ -16,24 +16,24 @@
...
         "webpack:prod": "npm-run-all webpack:production:*",
-        "webpack:analyze": "npm run webpack:production:web -- --env.presets analyze",
+        "webpack:analyze": "yarn run webpack:production:web -- --env.presets analyze",
```
Comment (mindy) [inline package.json:30]:
```suggestion
        "webpack:analyze": "yarn run webpack:production:web --env.presets analyze",
```

File: package.json (hunk 1 above)
Comment (mindy) [inline package.json:19]:
```suggestion
        "debug:development:web": "yarn run debug --env.presets web --env.presets svg --env.mode development",
```

**Pull request merged**